lib/services/local/git_server.go

```diff
@@ -128,3 +130,44 @@ func (s *GitServerService) ListGitServers(ctx context.Context, pageSize int, pag
 	}
 	return items, next, nil
 }
+
+func newGitServerParser() *gitServerParser {
+	return &gitServerParser{
+		baseParser: newBaseParser(backend.NewKey(gitServerPrefix)),
+	}
+}
+
+type gitServerParser struct {
+	baseParser
+}
+
+func (p *gitServerParser) parse(event backend.Event) (types.Resource, error) {
+	switch event.Type {
+	case types.OpDelete:
+		name := event.Item.Key.TrimPrefix(backend.NewKey(gitServerPrefix)).String()
+		if name == "" {
+			return nil, trace.NotFound("failed parsing %v", event.Item.Key.String())
+		}
+
+		return &types.ResourceHeader{
+			Kind:    types.KindGitServer,
+			Version: types.V2,
+			Metadata: types.Metadata{
+				Name:      strings.TrimPrefix(name, backend.SeparatorString),
```

Comment:
nit: we can take advantage of the key components instead of doing string manipulation

```suggestion
		components := event.Item.Key.Components()
		if len(components) < 2 {
			return nil, trace.NotFound("failed parsing %s", event.Item.Key)
		}

		return &types.ResourceHeader{
			Kind:    types.KindGitServer,
			Version: types.V2,
			Metadata: types.Metadata{
				Name:      components[1],
```

Reply:
nice. i think i copied it from somewhere but this is way better